composer/datasets/ade20k_hparams.py

```diff
@@ -176,15 +179,61 @@ class StreamingADE20kHparams(DatasetHparams):
     ignore_background: bool = hp.optional('If true, ignore the background class in training loss', default=True)
 
     def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHparams) -> DataSpec:
-        dataset = StreamingADE20k(remote=self.remote,
-                                  local=self.local,
-                                  split=self.split,
-                                  shuffle=self.shuffle,
-                                  base_size=self.base_size,
-                                  min_resize_scale=self.min_resize_scale,
-                                  max_resize_scale=self.max_resize_scale,
-                                  final_size=self.final_size,
-                                  batch_size=batch_size)
+        if self.version == 1:
+            dataset = StreamingADE20k(remote=self.remote,
+                                      local=self.local,
+                                      split=self.split,
+                                      shuffle=self.shuffle,
+                                      base_size=self.base_size,
+                                      min_resize_scale=self.min_resize_scale,
+                                      max_resize_scale=self.max_resize_scale,
+                                      final_size=self.final_size,
+                                      batch_size=batch_size)
+        elif self.version == 2:
+            try:
+                from streaming.vision import ADE20K
+            except ImportError as e:
+                raise MissingConditionalImportError(extra_deps_group='streaming',
+                                                    conda_package='mosaicml-streaming') from e
+            # Define data transformations based on data split
+            if self.split == 'train':
+                both_transforms = torch.nn.Sequential(
```

Comment:
should this be `torchvision.transforms.Compose`

Reply:
Generally we don’t use `torch.nn.Sequential()` for non-scriptable transformation and `transforms.ToTensor()` is not a scriptable transformation and we don’t have that in ADE20K. So, it's fine to use  `torch.nn.Sequential()` here.  Also, the code structure is similar to existing [ade20k.py](https://github.com/mosaicml/composer/blob/dev/composer/datasets/ade20k.py#L339).